Storage:  query string order dependency in 'test_create_bucket_with_string_confict' unit test.

From [this CI failure](https://source.cloud.google.com/results/invocations/96bf63b8-01b5-4c0d-b9fd-f737e3ba624e/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fstorage/log):

```python
______________ TestClient.test_create_bucket_with_string_conflict ______________

self = <tests.unit.test_client.TestClient testMethod=test_create_bucket_with_string_conflict>

    def test_create_bucket_with_string_conflict(self):
        from google.cloud.exceptions import Conflict

        project = "PROJECT"
        user_project = "USER_PROJECT"
        other_project = "OTHER_PROJECT"
        credentials = _make_credentials()
        client = self._make_one(project=project, credentials=credentials)

        bucket_name = "bucket-name"
        URI = "/".join(
            [
                client._connection.API_BASE_URL,
                "storage",
                client._connection.API_VERSION,
                "b?project=%s&userProject=%s" % (other_project, user_project),
            ]
        )
        data = {"error": {"message": "Conflict"}}
        json_expected = {"name": bucket_name}
        http = _make_requests_session(
            [_make_json_response(data, status=http_client.CONFLICT)]
        )
        client._http_internal = http

        with self.assertRaises(Conflict):
            client.create_bucket(
                bucket_name, project=other_project, user_project=user_project
            )

        http.request.assert_called_once_with(
>           method="POST", url=URI, data=mock.ANY, headers=mock.ANY
        )

tests/unit/test_client.py:551:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.nox/unit-3-5/lib/python3.5/site-packages/mock/mock.py:957: in assert_called_once_with
    return self.assert_called_with(*args, **kwargs)
.nox/unit-3-5/lib/python3.5/site-packages/mock/mock.py:944: in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

value = None, from_value = None

>   ???
E   Expected: request(data=<ANY>, headers=<ANY>, method='POST', url='https://storage.googleapis.com/storage/v1/b?project=OTHER_PROJECT&userProject=USER_PROJECT')
E   Actual: request(data='{"name": "bucket-name"}', headers={'Content-Type': 'application/json', 'X-Goog-API-Client': 'gcloud-python/1.22.0  gl-python/3.5.3 gax/1.14.3 gccl/1.22.0', 'User-Agent': 'gcloud-python/1.22.0  gl-python/3.5.3 gax/1.14.3 gccl/1.22.0', 'Accept-Encoding': 'gzip'}, method='POST', url='https://storage.googleapis.com/storage/v1/b?userProject=USER_PROJECT&project=OTHER_PROJECT')
```

--- a/storage/tests/unit/test_client.py
+++ b/storage/tests/unit/test_client.py
@@ -517,7 +517,14 @@ class TestClient(unittest.TestCase):
             method="GET", url=URI, data=mock.ANY, headers=mock.ANY
         )
 
-    def test_create_bucket_w_string_conflict(self):
+    def test_create_bucket_w_missing_client_project(self):
+        credentials = _make_credentials()
+        client = self._make_one(project=None, credentials=credentials)
+
+        with self.assertRaises(ValueError):
+            client.create_bucket("bucket")
+
+    def test_create_bucket_w_conflict(self):
         from google.cloud.exceptions import Conflict
 
         project = "PROJECT"
@@ -552,13 +559,6 @@ class TestClient(unittest.TestCase):
         )
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(json_expected, json.loads(json_sent))
-
-    def test_create_bucket_w_missing_client_project(self):
-        credentials = _make_credentials()
-        client = self._make_one(project=None, credentials=credentials)
-
-        with self.assertRaises(ValueError):
-            client.create_bucket("bucket")
 
     def test_create_bucket_w_predefined_acl_invalid(self):
         project = "PROJECT"

--- a/storage/tests/unit/test_client.py
+++ b/storage/tests/unit/test_client.py
@@ -532,33 +532,25 @@ class TestClient(unittest.TestCase):
         other_project = "OTHER_PROJECT"
         credentials = _make_credentials()
         client = self._make_one(project=project, credentials=credentials)
+        connection = _make_connection()
+        client._base_connection = connection
+        connection.api_request.side_effect = Conflict("testing")
 
         bucket_name = "bucket-name"
-        URI = "/".join(
-            [
-                client._connection.API_BASE_URL,
-                "storage",
-                client._connection.API_VERSION,
-                "b?project=%s&userProject=%s" % (other_project, user_project),
-            ]
-        )
-        data = {"error": {"message": "Conflict"}}
-        json_expected = {"name": bucket_name}
-        http = _make_requests_session(
-            [_make_json_response(data, status=http_client.CONFLICT)]
-        )
-        client._http_internal = http
+        data = {"name": bucket_name}
 
         with self.assertRaises(Conflict):
             client.create_bucket(
                 bucket_name, project=other_project, user_project=user_project
             )
 
-        http.request.assert_called_once_with(
-            method="POST", url=URI, data=mock.ANY, headers=mock.ANY
+        connection.api_request.assert_called_once_with(
+            method="POST",
+            path="/b",
+            query_params={"project": other_project, "userProject": user_project},
+            data=data,
+            _target_object=mock.ANY,
         )
-        json_sent = http.request.call_args_list[0][1]["data"]
-        self.assertEqual(json_expected, json.loads(json_sent))
 
     def test_create_bucket_w_predefined_acl_invalid(self):
         project = "PROJECT"

--- a/storage/tests/unit/test_client.py
+++ b/storage/tests/unit/test_client.py
@@ -589,100 +589,98 @@ class TestClient(unittest.TestCase):
         self.assertEqual(json_expected, json.loads(json_sent))
 
     def test_create_bucket_w_missing_client_project(self):
-        client = self._make_one(project=None)
+        credentials = _make_credentials()
+        client = self._make_one(project=None, credentials=credentials)
 
         with self.assertRaises(ValueError):
             client.create_bucket("bucket")
 
     def test_create_bucket_w_predefined_acl_invalid(self):
-        PROJECT = "PROJECT"
-        BUCKET_NAME = "bucket-name"
+        project = "PROJECT"
+        bucket_name = "bucket-name"
         credentials = _make_credentials()
-        client = self._make_one(project=PROJECT, credentials=credentials)
+        client = self._make_one(project=project, credentials=credentials)
 
         with self.assertRaises(ValueError):
-            client.create_bucket(BUCKET_NAME, predefined_acl="bogus")
+            client.create_bucket(bucket_name, predefined_acl="bogus")
 
     def test_create_bucket_w_predefined_acl_valid(self):
-        from google.cloud.storage.client import Client
+        project = "PROJECT"
+        bucket_name = "bucket-name"
+        data = {"name": bucket_name}
 
-        PROJECT = "PROJECT"
-        BUCKET_NAME = "bucket-name"
-        DATA = {"name": BUCKET_NAME}
-
-        client = Client(project=PROJECT)
-        connection = _make_connection(DATA)
+        credentials = _make_credentials()
+        client = self._make_one(project=project, credentials=credentials)
+        connection = _make_connection(data)
         client._base_connection = connection
-        bucket = client.create_bucket(BUCKET_NAME, predefined_acl="publicRead")
+        bucket = client.create_bucket(bucket_name, predefined_acl="publicRead")
 
         connection.api_request.assert_called_once_with(
             method="POST",
             path="/b",
-            query_params={"project": PROJECT, "predefinedAcl": "publicRead"},
-            data=DATA,
+            query_params={"project": project, "predefinedAcl": "publicRead"},
+            data=data,
             _target_object=bucket,
         )
 
     def test_create_bucket_w_predefined_default_object_acl_invalid(self):
-        PROJECT = "PROJECT"
-        BUCKET_NAME = "bucket-name"
+        project = "PROJECT"
+        bucket_name = "bucket-name"
 
         credentials = _make_credentials()
-        client = self._make_one(project=PROJECT, credentials=credentials)
+        client = self._make_one(project=project, credentials=credentials)
 
         with self.assertRaises(ValueError):
-            client.create_bucket(BUCKET_NAME, predefined_default_object_acl="bogus")
+            client.create_bucket(bucket_name, predefined_default_object_acl="bogus")
 
     def test_create_bucket_w_predefined_default_object_acl_valid(self):
-        from google.cloud.storage.client import Client
+        project = "PROJECT"
+        bucket_name = "bucket-name"
+        data = {"name": bucket_name}
 
-        PROJECT = "PROJECT"
-        BUCKET_NAME = "bucket-name"
-        DATA = {"name": BUCKET_NAME}
-
-        client = Client(project=PROJECT)
-        connection = _make_connection(DATA)
+        credentials = _make_credentials()
+        client = self._make_one(project=project, credentials=credentials)
+        connection = _make_connection(data)
         client._base_connection = connection
         bucket = client.create_bucket(
-            BUCKET_NAME, predefined_default_object_acl="publicRead"
+            bucket_name, predefined_default_object_acl="publicRead"
         )
 
         connection.api_request.assert_called_once_with(
             method="POST",
             path="/b",
             query_params={
-                "project": PROJECT,
+                "project": project,
                 "predefinedDefaultObjectAcl": "publicRead",
             },
-            data=DATA,
+            data=data,
             _target_object=bucket,
         )
 
     def test_create_bucket_w_explicit_location(self):
-        from google.cloud.storage.client import Client
-
-        PROJECT = "PROJECT"
-        BUCKET_NAME = "bucket-name"
-        LOCATION = "us-central1"
-        DATA = {"location": LOCATION, "name": BUCKET_NAME}
+        project = "PROJECT"
+        bucket_name = "bucket-name"
+        location = "us-central1"
+        data = {"location": location, "name": bucket_name}
 
         connection = _make_connection(
-            DATA, "{'location': 'us-central1', 'name': 'bucket-name'}"
+            data, "{'location': 'us-central1', 'name': 'bucket-name'}"
         )
 
-        client = Client(project=PROJECT)
+        credentials = _make_credentials()
+        client = self._make_one(project=project, credentials=credentials)
         client._base_connection = connection
 
-        bucket = client.create_bucket(BUCKET_NAME, location=LOCATION)
+        bucket = client.create_bucket(bucket_name, location=location)
 
         connection.api_request.assert_called_once_with(
             method="POST",
             path="/b",
-            data=DATA,
+            data=data,
             _target_object=bucket,
-            query_params={"project": "PROJECT"},
+            query_params={"project": project},
         )
-        self.assertEqual(bucket.location, LOCATION)
+        self.assertEqual(bucket.location, location)
 
     def test_create_bucket_w_string_success(self):
         from google.cloud.storage.bucket import Bucket

--- a/storage/tests/unit/test_client.py
+++ b/storage/tests/unit/test_client.py
@@ -517,7 +517,7 @@ class TestClient(unittest.TestCase):
             method="GET", url=URI, data=mock.ANY, headers=mock.ANY
         )
 
-    def test_create_bucket_with_string_conflict(self):
+    def test_create_bucket_w_string_conflict(self):
         from google.cloud.exceptions import Conflict
 
         project = "PROJECT"
@@ -553,7 +553,7 @@ class TestClient(unittest.TestCase):
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(json_expected, json.loads(json_sent))
 
-    def test_create_bucket_with_object_conflict(self):
+    def test_create_bucket_w_object_conflict(self):
         from google.cloud.exceptions import Conflict
         from google.cloud.storage.bucket import Bucket
 
@@ -588,13 +588,13 @@ class TestClient(unittest.TestCase):
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(json_expected, json.loads(json_sent))
 
-    def test_create_w_missing_client_project(self):
+    def test_create_bucket_w_missing_client_project(self):
         client = self._make_one(project=None)
 
         with self.assertRaises(ValueError):
             client.create_bucket("bucket")
 
-    def test_create_w_predefined_acl_invalid(self):
+    def test_create_bucket_w_predefined_acl_invalid(self):
         PROJECT = "PROJECT"
         BUCKET_NAME = "bucket-name"
         credentials = _make_credentials()
@@ -603,7 +603,7 @@ class TestClient(unittest.TestCase):
         with self.assertRaises(ValueError):
             client.create_bucket(BUCKET_NAME, predefined_acl="bogus")
 
-    def test_create_w_predefined_acl_valid(self):
+    def test_create_bucket_w_predefined_acl_valid(self):
         from google.cloud.storage.client import Client
 
         PROJECT = "PROJECT"
@@ -623,7 +623,7 @@ class TestClient(unittest.TestCase):
             _target_object=bucket,
         )
 
-    def test_create_w_predefined_default_object_acl_invalid(self):
+    def test_create_bucket_w_predefined_default_object_acl_invalid(self):
         PROJECT = "PROJECT"
         BUCKET_NAME = "bucket-name"
 
@@ -633,7 +633,7 @@ class TestClient(unittest.TestCase):
         with self.assertRaises(ValueError):
             client.create_bucket(BUCKET_NAME, predefined_default_object_acl="bogus")
 
-    def test_create_w_predefined_default_object_acl_valid(self):
+    def test_create_bucket_w_predefined_default_object_acl_valid(self):
         from google.cloud.storage.client import Client
 
         PROJECT = "PROJECT"
@@ -658,7 +658,7 @@ class TestClient(unittest.TestCase):
             _target_object=bucket,
         )
 
-    def test_create_w_explicit_location(self):
+    def test_create_bucket_w_explicit_location(self):
         from google.cloud.storage.client import Client
 
         PROJECT = "PROJECT"
@@ -684,7 +684,7 @@ class TestClient(unittest.TestCase):
         )
         self.assertEqual(bucket.location, LOCATION)
 
-    def test_create_bucket_with_string_success(self):
+    def test_create_bucket_w_string_success(self):
         from google.cloud.storage.bucket import Bucket
 
         project = "PROJECT"
@@ -716,7 +716,7 @@ class TestClient(unittest.TestCase):
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(json_expected, json.loads(json_sent))
 
-    def test_create_bucket_with_object_success(self):
+    def test_create_bucket_w_object_success(self):
         from google.cloud.storage.bucket import Bucket
 
         project = "PROJECT"

--- a/storage/tests/unit/test_client.py
+++ b/storage/tests/unit/test_client.py
@@ -553,41 +553,6 @@ class TestClient(unittest.TestCase):
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(json_expected, json.loads(json_sent))
 
-    def test_create_bucket_w_object_conflict(self):
-        from google.cloud.exceptions import Conflict
-        from google.cloud.storage.bucket import Bucket
-
-        project = "PROJECT"
-        other_project = "OTHER_PROJECT"
-        credentials = _make_credentials()
-        client = self._make_one(project=project, credentials=credentials)
-
-        bucket_name = "bucket-name"
-        bucket_obj = Bucket(client, bucket_name)
-        URI = "/".join(
-            [
-                client._connection.API_BASE_URL,
-                "storage",
-                client._connection.API_VERSION,
-                "b?project=%s" % (other_project,),
-            ]
-        )
-        data = {"error": {"message": "Conflict"}}
-        http = _make_requests_session(
-            [_make_json_response(data, status=http_client.CONFLICT)]
-        )
-        client._http_internal = http
-
-        with self.assertRaises(Conflict):
-            client.create_bucket(bucket_obj, project=other_project)
-
-        http.request.assert_called_once_with(
-            method="POST", url=URI, data=mock.ANY, headers=mock.ANY
-        )
-        json_expected = {"name": bucket_name}
-        json_sent = http.request.call_args_list[0][1]["data"]
-        self.assertEqual(json_expected, json.loads(json_sent))
-
     def test_create_bucket_w_missing_client_project(self):
         credentials = _make_credentials()
         client = self._make_one(project=None, credentials=credentials)

